src/trustyai/metrics/distance.py

```diff
@@ -0,0 +1,116 @@
+""""Distance metrics"""
+# pylint: disable = import-error
+from dataclasses import dataclass
+from typing import List, Optional, Union, Callable
+
+from org.kie.trustyai.metrics.language.distance import (
+    Levenshtein as _Levenshtein,
+    LevenshteinResult as _LevenshteinResult,
+    LevenshteinCounters as _LevenshteinCounters,
+)
+from opennlp.tools.tokenize import Tokenizer
+import numpy as np
+import matplotlib.pyplot as plt
+from trustyai import _default_initializer  # pylint: disable=unused-import
+
+
+@dataclass
+class LevenshteinCounters:
+    """LevenshteinCounters Counters"""
+
+    substitutions: int
+    insertions: int
+    deletions: int
+    correct: int
+
+    @staticmethod
+    def convert(result: _LevenshteinCounters):
+        """Converts a Java LevenshteinCounters to a Python LevenshteinCounters"""
+        return LevenshteinCounters(
+            substitutions=result.getSubstitutions(),
+            insertions=result.getInsertions(),
+            deletions=result.getDeletions(),
+            correct=result.getCorrect(),
+        )
+
+
+@dataclass
+class LevenshteinResult:
+    """Levenshtein Result"""
+
+    distance: float
+    counters: LevenshteinCounters
+    matrix: np.ndarray
+    reference: List[str]
+    hypothesis: List[str]
+
+    @staticmethod
+    def convert(result: _LevenshteinResult):
+        """Converts a Java LevenshteinResult to a Python LevenshteinResult"""
+        distance = result.getDistance()
+        counters = LevenshteinCounters.convert(result.getCounters())
+        data = result.getDistanceMatrix().getData()
+        numpy_array = np.array(data)[1:, 1:]
+        reference = result.getReferenceTokens()
+        hypothesis = result.getHypothesisTokens()
+
+        return LevenshteinResult(
+            distance=distance,
+            counters=counters,
+            matrix=numpy_array,
+            reference=reference,
+            hypothesis=hypothesis,
+        )
+
+    def plot(self):
```

Comment:
Minor thing and maybe it is already like that in the rest of the library. I don't like too much having plotting logic (and dependency) in the _core_ of the library.
I would probably prefer something like `trustyai[plot]` if possible.
Btw this is a side comment, if the team agree, we can create a followup ticket

Reply:
I've created a placeholder ticket to track this discussion

- https://github.com/trustyai-explainability/trustyai-explainability-python/issues/192